### PR TITLE
Closes #1501 Save all datasets in templates as .rds instead of .rda

### LINKED ISSUE #1501: Save datasets in templates as `.rds` files
Currently we save the datasets generated within our templates as `.rda` files. This has histroical reasons: we used to run the templates to generate the ADaM datasets that used to be part of {admiral}. However, that's no longer the case. `.rda` files have the disadvantage that they write their contents directly to the global environment without the user having control over the name of the create variable. `.rds` files do not suffer from this downside.

### PR DESCRIPTION
**Pull request by nkr821**
Thank you for your Pull Request! We have developed this task checklist from the [Development Process Guide](https://pharmaverse.github.io/admiraldev/main/articles/development_process.html) to help with the final steps of the process. Completing the below tasks helps to ensure our reviewers can maximize their time on your code as well as making sure the admiral codebase remains robust and consistent.   

Please check off each taskbox as an acknowledgment that you completed the task or check off that it is not relevant to your Pull Request. This checklist is part of the Github Action workflows and the Pull Request will not be merged into the `devel` branch until you have checked off each task.

- [x] Place Closes #<insert_issue_number> into the beginning of your Pull Request Title (Use Edit button in top-right if you need to update)
- [ ] Code is formatted according to the [tidyverse style guide](https://style.tidyverse.org/). Run `styler::style_file()` to style R and Rmd files
- [ ] Updated relevant unit tests or have written new unit tests - See [Unit Test Guide](https://pharmaverse.github.io/admiraldev/main/articles/unit_test_guidance.html#writing-unit-tests-in-admiral-)
- [ ] If you removed/replaced any function and/or function parameters, did you fully follow the [deprecation guidance](https://pharmaverse.github.io/admiraldev/main/articles/programming_strategy.html#deprecation-1)?
- [ ] Update to all relevant roxygen headers and examples 
- [ ] Run `devtools::document()` so all `.Rd` files in the `man` folder and the `NAMESPACE` file in the project root are updated appropriately
- [ ] Address any updates needed for vignettes and/or templates
- [ ] Update `NEWS.md` if the changes pertain to a user-facing function (i.e. it has an `@export` tag) or documentation aimed at users (rather than developers)
- [ ] Build admiral site `pkgdown::build_site()` and check that all affected examples are displayed correctly and that all new functions occur on the "[Reference](https://pharmaverse.github.io/admiral/reference/index.html)" page. 
- [ ] Address or fix all lintr warnings and errors - `lintr::lint_package()`
- [ ] Run `R CMD check` locally and address all errors and warnings - `devtools::check()`
- [x] Link the issue in the Development Section on the right hand side.
- [ ] Address all merge conflicts and resolve appropriately 
- [ ] Pat yourself on the back for a job well done! Much love to your accomplishment!

### CODE REVIEW

**Review by thomas-neitmann — CHANGES_REQUESTED** [review]

File: inst/templates/ad_adae.R [hunk 1]
```diff
@@ -135,4 +135,4 @@ adae <- adae %>%
 # Save output ----
 
 dir <- tempdir() # Change to whichever directory you want to save the dataset in
-save(adae, file = file.path(dir, "adae.rda"), compress = "bzip2")
+save(adae, file = file.path(dir, "adae.rds"), compress = "bzip2")
```
[the remark this answers is not in the document]
Comment (nkr821) [reply inst/templates/ad_adae.R:138]: saveRDS() does have a compress argument. I'll make the needed changes today

**Review by thomas-neitmann — CHANGES_REQUESTED** [review]
Thanks a lot for this update @nkr821! Can you also add an entry in the `NEWS.md` file?

**Review by thomas-neitmann — APPROVED** [review]
Thanks a lot @nkr821!

**Comment by thomas-neitmann** [comment]
@nkr821 As a final step can you also follow these instructions such that the "Spell check" won't fail anymore:

```
Warning message:
In if (!is.na(exclusions)) removed <- lapply(Sys.glob(exclusions),  :
  the condition has length > 1 and only the first element will be used
  WORD  FOUND IN
rda   NEWS.md:25
rds   NEWS.md:25

Number of misspelled words: 2
You may correct the spellings of the words above or add them to the "inst/WORDLIST" file by running spelling::update_wordlist()
```

**Comment by bms63** [comment]
Hi @nkr821 can you fix these merge and Github Actions please.  Would like to get this merged into devel.  Thanks!!

**Comment by nkr821** [comment]
Hi @bms63, I'm still pretty new to Admiral, so I'm not sure what specifically needs to be done. I updated the WORDLIST file with the words failing the spellcheck, so I thought that would fix the problem, but I guess it is still not working. Any ideas on what to do next?

File: NEWS.md [hunk 2]
```diff
@@ -25,6 +25,8 @@ which controls if time is considered for the condition if `start_date` is after
 - Function `derive_vars_merged()` had an improperly formatted error message 
 which has been corrected (#1473)
 
+- Templates now save datasets as .rds instead of .rda (#1501)
```
Comment (bms63) [inline NEWS.md:28]:
```suggestion
- Templates now save datasets as `.rds` instead of `.rda` (#1501)
```

Comment (bms63) [reply NEWS.md:28]: Perhaps the backticks will fix this issue for us?

**Comment by bms63** [comment]
It worked!  Yay!  CAn you remove rda and rds from spell check list unless it is used somewhere else??  Also click the Update branch button so the latest devel updates come into your feature branch.

**Comment by nkr821** [comment]
Yay! Sounds good; I'll do that now

**Comment by nkr821** [comment]
@bms63 when I went to update the wordlist again, I started to get strange commit errors after making the change, and now my local branch is a total disaster, haha. If that's the only thing left to do for this issue, could we just leave them in there for now and I can update the wordlist on the next issue I work on? Thanks!

**Comment by bms63** [comment]
> @bms63 when I went to update the wordlist again, I started to get strange commit errors after making the change, and now my local branch is a total disaster, haha. If that's the only thing left to do for this issue, could we just leave them in there for now and I can update the wordlist on the next issue I work on? Thanks!

All good.  I fixed it up.  This is your first PR to admiral!!?? Congrats!

**Comment by nkr821** [comment]
Only my second PR. Thanks, @bms63 for your help!

**Pull request merged**